javascript/ql/lib/semmle/javascript/security/dataflow/TaintedFormatStringSpecific.qll

```diff
@@ -0,0 +1,6 @@
+/**
+ * Provides JS-specific imports needed for `TaintedFormatStringQuery` and `TaintedFormatStringCustomizations`.
+ */
+
+import javascript
+import semmle.javascript.security.dataflow.DOM
```

Comment:
I think this `import semmle.javascript.security.dataflow.DOM` is redundant.  
But that's fine.  
(`ql/redundant-import` anyone?)

Reply:
A query for redundant imports would be great! Even better would be a compiler warning delivered via LSP.

Reply:
I don't think it belongs as a compiler warning.  
In the above case both of the imports eventually import each other, so deleting any one of them will solve it.  

And do you want a compiler warning on both for that? I think that feels wrong, and so does the arbitrary choice of warning on the second.  
And sometimes (redundant) imports and their order can be important (I'm not sure why, but I think the CPP team can tell stories).   

In QL-for-QL: arbitary choices and lingering warnings are fine, so I think it should be a query.  
(But I want to figure out how we'll get support for parameterized modules before I start more QL-for-QL query writing).